### DISCUSSION TITLE
.chezmoi.toml.tmpl not processed

### DISCUSSION THREAD

**Discussion by fslef** [twpayne/chezmoi#2664] [Q&A] [answered]
Hi everyone 👋🏻

I an trying to use the templating feature.

I've created a .chezmoi.toml.tmpl template file ([content available here](https://github.com/fslef/dotfiles/blob/main/.chezmoi.toml.tmpl))

When a new chez moi instance is deployed, or when I'm running the chezmoi init on an existing folder, the template file is not processed and the .chezmoi.toml is not created.

I tried multiple way to troubleshoot it but I have no ideas what's going wrong.
Any advices to help me to troubleshoot it ?

**Comment by Lockszmith-GH** [comment, marked as answer]
Hi @fslef,

Thanks for sharing the link to your repo, that helps.
The `.chezmoi.toml.tmpl` needs to be in your **chezmoi root**, and you have a `.chezmoihome` defining the root is `home`.
Move the file to `home` and you should be good.

**Reply by fslef** [reply]
Hi @Lockszmith-GH.

Thanks a lot for your help.
I confirm that it work after moving the `.chezmoi.toml.tmpl` file to the `home` folder.

Does it means that on the root folder I should only keep the `.chezmoihome` file and I should then move each other `.chezmoixxx` files to the `home` folder as well (such as the `.chezmoiignore`)?

**Reply by twpayne** [reply]
Yes, if you're using `.chezmoiroot` then only `.chezmoiroot` and `.chezmoiversion` should be in the root folder. For a quick view of what it should look like, check out [my dotfiles](https://github.com/twpayne/dotfiles).

**Reply by fslef** [reply]
All right. Thanks a lot for the clarification.
